dotnet/src/SemanticKernel.Abstractions/AI/ChatCompletion/IChatMessage.cs

```diff
@@ -0,0 +1,9 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+namespace Microsoft.SemanticKernel.AI.ChatCompletion;
+
+public interface IChatMessage
```

Comment:
As a general note for API design in .NET: Utilizing interfaces gives an "I can do" methodology where as things like abstract classes represent an "I am". In addition, once SK goes GA you'll be bound to these interfacers and if you ever want to change them you'll need to make an `IChatMessage2` in order to expand. Alternatively, if you use an abstract class you can simply add new members as virtual.

Gist: Interfaces have a place in .NET but have a lot of flaws. For instance, all of Roslyn and a lot of ASP.NET migrated away from interface representations specifically for this reason.